Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1797,16 +1797,18 @@ private string GetVideoArguments(StreamState state, int startNumber, bool isEven
 
         var args = "-codec:v:0 " + codec;
 
+        var requestedRange = state.GetRequestedRangeTypes(codec);
+
         if (string.Equals(state.ActualOutputVideoCodec, "h265", StringComparison.OrdinalIgnoreCase)
             || string.Equals(state.ActualOutputVideoCodec, "hevc", StringComparison.OrdinalIgnoreCase)
             || string.Equals(codec, "h265", StringComparison.OrdinalIgnoreCase)
             || string.Equals(codec, "hevc", StringComparison.OrdinalIgnoreCase))
         {
             if (EncodingHelper.IsCopyCodec(codec)
-                && (state.VideoStream.VideoRangeType == VideoRangeType.DOVI
-                    || string.Equals(state.VideoStream.CodecTag, "dovi", StringComparison.OrdinalIgnoreCase)
-                    || string.Equals(state.VideoStream.CodecTag, "dvh1", StringComparison.OrdinalIgnoreCase)
-                    || string.Equals(state.VideoStream.CodecTag, "dvhe", StringComparison.OrdinalIgnoreCase)))
+                && ((state.VideoStream.VideoRangeType == VideoRangeType.DOVI && requestedRange.Contains(VideoRangeType.DOVI.ToString(), StringComparison.OrdinalIgnoreCase))
+                    || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithHDR10 && requestedRange.Contains(VideoRangeType.DOVIWithHDR10.ToString(), StringComparison.OrdinalIgnoreCase))
+                    || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithHLG && requestedRange.Contains(VideoRangeType.DOVIWithHLG.ToString(), StringComparison.OrdinalIgnoreCase))
+                    || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithSDR && requestedRange.Contains(VideoRangeType.DOVIWithSDR.ToString(), StringComparison.OrdinalIgnoreCase))))
```

Comment:
```suggestion
                && ((state.VideoStream.VideoRangeType == VideoRangeType.DOVI && haveDOVI)
                     || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithHDR10 && haveDOVIWithHDR10)
                     || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithHLG && haveDOVIWithHLG)
                     || (state.VideoStream.VideoRangeType == VideoRangeType.DOVIWithSDR && haveDOVIWithSDR)))
```

Each line is already too long, Extract the contains* to vars to increase readability.